Update apiDefinition.swagger.json

jokano12 opened the pull request:
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

jokano12 committed:
Update apiDefinition.swagger.json

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -2322,7 +2584,7 @@
         ]
       }
-    },
+    ,
```
There is a missing closing curly bracket here

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -2395,6 +2657,80 @@
           }
         ]
       }
+    },
```
There is an extra closing curly bracket here

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -2395,6 +2657,80 @@
+        "operationId": "List-All-Organizations",
+        "parameters": [
+          {
+            "name": "Page Number",
```
The `name` parameter here should to be changed to `id`

lordlydon replied:
Also here
For the Page Number, please add `"x-ms-visibility": "advanced"` and `"default": 1`

jokano12 replied:
I removed the whole action "List all Organizations" as there are still internal decisions needed to be made.

lordlydon replied:
@RjImpx, is this going to be sent as part of a different PR, or should we wait until "List all Organizations" is ready to merge PR?

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -2395,6 +2657,80 @@
+            "required": true,
+            "type": "integer",
+            "default": 1,
+            "description": "Page  Number.",
```
There is an extra space in the description

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -3781,6 +4117,1776 @@
+      "properties": {
+        "primary": {
+          "type": "boolean",
+          "description": "Primary Organization - Emails - Is Primary.",
```
Please remove the `Primary Organization - Emails -` here as this is not a nested object.

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -3781,6 +4117,1776 @@
+        },
+        "address": {
+          "type": "string",
+          "description": "Primary Organization - Emails - Address.",
```
Please remove prefix here too

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -3781,6 +4117,1776 @@
+        },
+        "type": {
+          "type": "string",
+          "description": "Primary Organization - Emails - Type.",
```
Please remove prefix here too

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -3781,6 +4117,1776 @@
+        },
+        "showInDirectory": {
+          "type": "boolean",
+          "description": "Primary Organization - Emails - Show In Directory.",
```
Please remove prefix here too

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -80,6 +80,268 @@
+            "x-ms-url-encoding": "single"
+          },
+          {
+            "name": "Page Number",
```
For the Page Number, please add `"x-ms-visibility": "advanced"` and `"default": 1`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -80,6 +80,268 @@
+            "x-ms-url-encoding": "single"
+          },
+          {
+            "name": "Page Number",
```
Also here
For the Page Number, please add `"x-ms-visibility": "advanced"` and `"default": 1`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -2395,6 +2657,80 @@
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "Organization Name.",
```
Please update the description and summary

lordlydon reviewed (changes requested):
Hi @jokano12,
I added a couple of comments, please review.

jokano12 committed:
Update apiDefinition.swagger.json

I removed the action "List all Organization" since there are internal decisions that are needing to be made.

RjImpx commented:
It can be part of a different PR so we can move forward.

[Visit Billhighway]
Raaj Jeeth
VP, Product Development
Phone: 877.486.8220
[https://cdn.billhighway.com/images/bh-gb+signature/facebook.png]<https://www.facebook.com/impexium>[https://cdn.billhighway.com/images/bh-gb+signature/twitter.png]<https://twitter.com/impexium>[https://cdn.billhighway.com/images/bh-gb+signature/linkedin.png]<https://www.linkedin.com/company/impexium>[https://cdn.billhighway.com/images/bh-gb+signature/rss.png]<https://www.impexium.com/>

From: William Llanos ***@***.***>
Sent: Tuesday, November 2, 2021 4:43 PM
To: RjImpx/PowerPlatformConnectors ***@***.***>
Cc: Raaj Jeeth ***@***.***>; Mention ***@***.***>
Subject: Re: [RjImpx/PowerPlatformConnectors] Update apiDefinition.swagger.json (PR #17)

[EXTERNAL] This message originated outside of Billhighway/Greekbill/Impexium.

@LordLydon commented on this pull request.

________________________________

In certified-connectors/Impexium/apiDefinition.swagger.json<https://github.com/RjImpx/PowerPlatformConnectors/pull/17#discussion_r741447418>:

> +          "Organizations"

+        ],

+        "responses": {

+          "200": {

+            "description": "Success",

+            "schema": {

+              "$ref": "#/definitions/OrganizationData"

+            }

+          }

+        },

+        "summary": "List All Organizations",

+        "description": "List All Organizations",

+        "operationId": "List-All-Organizations",

+        "parameters": [

+          {

+            "name": "Page Number",

@RjImpx<https://github.com/RjImpx>, is this going to be sent as part of a different PR, or should we wait until "List all Organizations" is ready to merge PR?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/RjImpx/PowerPlatformConnectors/pull/17#discussion_r741447418>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ARLDTJ7RUTNWJOBOZ5TENRDUKBEMPANCNFSM5G5XSPHA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

lordlydon reviewed (approved):
Looks good

jokano12 commented:
Actions(5):
Get Committee Members by ID or Code
Find Members (Individuals) by First and Last Name
Add/Update address to Individual
Add Email to Individual
List All organizations
